app/helpers/ems_physical_infra_helper/textual_summary.rb

```diff
@@ -62,9 +62,13 @@ def textual_physical_servers
     h
   end
 
-  def textual_hosts
+  def textual_physical_servers_with_host
     count_of_host_relationships = (@record.physical_servers.select { |server| !server.host.nil? }).length
-    {:label => _("Hosts"), :icon => "pficon pficon-screen", :value => count_of_host_relationships}
+    h = {:label => _("Physical Servers with Host"), :icon => "pficon pficon-server", :value => count_of_host_relationships}
+    if count_of_host_relationships > 0
+      h[:link] = "/ems_physical_infra/#{@ems.id}?display=physical_servers&options=physical_servers_with_host"
```

Comment:
> > Can that be expressed as some (cheeper) AREL query that could result in an SQL count(...) query?

Good point.
Currently the association between `physical_servers` and `hosts` exists with `hosts.physical_server_id` (so it's from `hosts` -> `physical_servers`)

I think we are missing an association from `physical_servers` to `hosts` 
( `physical_servers` -> `hosts`)

Once we have that, the query should look much simpler (and will be cheaper).